src/components/MainNavBar/MainNavBar.tsx

```diff
@@ -17,16 +17,22 @@ const MainNavBar = () => {
   const [toastMessage, setToastMessage] = useState<string>("");
   const [toastType, setToastType] = useState<string>("danger");
   const terminologyServiceApi = useTerminologyServiceApi();
-  const [isLoggedInToUMLS, setIsLoggedInToUMLS] = useState<boolean>(false);
+  const [isLoggedInToUMLS, setIsLoggedInToUMLS] = useState<boolean>(undefined);
 
-  useEffect(() => {
-    terminologyServiceApi
-      .checkLogin()
-      .then((value) => {
+  const checkLoginToUMLS = async () => {
+    const loginStatus = await Promise.resolve(
+      terminologyServiceApi.checkLogin().then((value) => {
         setIsLoggedInToUMLS(true);
       })
-      .catch((err) => {});
-  }, []);
+    );
+    return loginStatus;
+  };
+  const { authState } = useOktaAuth();
+  useEffect(() => {
+    if (authState?.isAuthenticated && !isLoggedInToUMLS) {
+      checkLoginToUMLS().catch((err) => {});
```

Comment:
I think you can directly write the checkLoginToUMLS code directly in this useEffect , this kinda removes the duplicate code

Reply:
This is done, thanks so much Prateek!!!